### ISSUE TITLE
Support Confluent Schema Registry

### ISSUE THREAD

**Issue by jeqo** [prestosql/presto#2105]
[Confluent Schema Registry](https://github.com/confluentinc/schema-registry) is commonly used to store Avro schemas, and reduce Kafka records size by storing an ID instead of the schema on the record message.

Currently, only Avro files are supported to decode Kafka records.

I'd like to propose adding support for Schema Registry on the Avro decoder.

**Comment by martint** [comment]
@elonazoulay, you've been looking into this, no?

**Comment by elonazoulay** [comment]
Yep, we will contribute what we have shortly, we also have a use case for this: we use the schema registry to supply metadata and publish keys and values using the String and Avro Kafka deserializers from confluent. 

@jeqo sounds very similar to our use case, once we put what we have up it would be great to collaborate on this, there might be things that are very specific to our use case, we can make them more general.

**Comment by elonazoulay** [comment]
@jeqo, here is the pull request: #2361 - still cleaning up the part where schema registry is used to infer the schema (i.e. without the need for json files).

**Comment by elonazoulay** [comment]
It looks like there is a lot of overlap between #2106 and #2361 :)

**Comment by cricket007** [comment]
Link https://github.com/prestodb/presto/issues/11354

**Comment by zhenik** [comment]
> Link [prestodb/presto#11354](https://github.com/prestodb/presto/issues/11354)
> 
> (edit) Why are there multiple repos?

facebook distribution and community opensource distribution. [Blog link explanation](https://blog.openbridge.com/what-is-facebook-presto-presto-database-or-prestodb-a-powerful-sql-query-engine-77d4c4a66d4)

**Comment by findepi** [comment]
@zhenik the blog you linked above is just a subjective commentary and IMO doesn't explain much.
@cricket007 You can see previous discussion under https://github.com/prestosql/presto/issues/380. 
If you have any doubts, I encourage you to reach out on our [community slack](https://prestosql.io/slack.html).


Let's keep the discussion here focused on the Confluent Schema Registry.

**Comment by cricket007** [comment]
Thanks for the links!

I only opened that issue after I saw Kafka reader was added
